packages/firebase_dynamic_links/ios/Classes/FLTFirebaseDynamicLinksPlugin.m

```diff
@@ -150,25 +150,61 @@ - (BOOL)checkForDynamicLink:(NSURL *)url {
 - (BOOL)application:(UIApplication *)application
             openURL:(NSURL *)url
             options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
-  return [self checkForDynamicLink:url];
+  [self checkForDynamicLink:url];
+  // results of this are ORed and NO doesn't affect other delegate interceptors' result
+  return NO;
 }
 
 - (BOOL)application:(UIApplication *)application
               openURL:(NSURL *)url
     sourceApplication:(NSString *)sourceApplication
            annotation:(id)annotation {
-  return [self checkForDynamicLink:url];
+  [self checkForDynamicLink:url];
+  // results of this are ORed and NO doesn't affect other delegate interceptors' result
+  return NO;
 }
 
 - (BOOL)application:(UIApplication *)application
     continueUserActivity:(NSUserActivity *)userActivity
-      restorationHandler:(void (^)(NSArray *))restorationHandler {
-  BOOL handled = [[FIRDynamicLinks dynamicLinks]
-      handleUniversalLink:userActivity.webpageURL
-               completion:^(FIRDynamicLink *_Nullable dynamicLink, NSError *_Nullable error) {
-                 [self onDeepLinkResult:dynamicLink error:error];
-               }];
-  return handled;
+      restorationHandler:
+#if defined(__IPHONE_12_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_12_0)
+          (nonnull void (^)(NSArray<id<UIUserActivityRestoring>> *_Nullable))restorationHandler {
+#else
+          (nonnull void (^)(NSArray *_Nullable))restorationHandler {
+#endif  // __IPHONE_12_0
+  __block BOOL retried = NO;
+
+  id completion =
+      ^(FIRDynamicLink *_Nullable dynamicLink, NSError *_Nullable error) {
+        if (!error && dynamicLink && dynamicLink.url) {
+          [self onDeepLinkResult:dynamicLink error:nil];
+        }
+      }
+      // Per Apple Tech Support, a network failure could occur when returning from background on
+      // iOS 12. https://github.com/AFNetworking/AFNetworking/issues/4279#issuecomment-447108981 So
+      // we'll retry the request once
+      if (error && !retried && [NSPOSIXErrorDomain isEqualToString:error.domain] &&
+          error.code == 53) {
+    retried = YES;
+    [[FIRDynamicLinks dynamicLinks] handleUniversalLink:userActivity.webpageURL
+                                             completion:completion];
+  }
+  // TODO: We could send this to Dart and maybe have a onDynamicLinkError listener but there's also
+  // a good chance the
+  // TODO: `userActivity.webpageURL` might not be for a Firebase dynamic link, which needs
+  // consideration - so we'll
+  // TODO: log this for now, logging will get picked up by Crashlytics automatically if its
+  // integrated.
```

Comment:
```suggestion
  // We could send this to Dart and maybe have a onDynamicLinkError stream but there's also
  // a good chance the `userActivity.webpageURL` might not be for a Firebase dynamic link, 
  // which needs consideration - so we'll log this for now, logging will get picked up by 
  // Crashlytics automatically if its integrated.
```

Reply:
@jherencia please can you apply this suggestion, then I can merge. For future PRs it's helpful to allow maintainers of the repository you're submitting your PR to to push changes to your branch.

Reply:
Done